src/string_map.c

```diff
@@ -166,6 +167,12 @@ rcutils_string_map_reserve(rcutils_string_map_t * string_map, size_t capacity)
     // note that realloc when the pointer is NULL is the same as malloc
     // note also that realloc will shrink the space if needed
 
+    // ensure that reallocate won't overflow capacity
+    if (capacity > (SIZE_MAX / sizeof(char *))) {
```

Comment:
I did not trace down all the callers of rcutils_string_map_reserve so it may not occur in practice today.  However, it may occur if the requested capacity is greater than SIZE_MAX / 4 or SIZE_MAX / 8, depending on the pointer size.  

I'll also note that glibc provides an explicit reallocarray() function that is the same as realloc() but includes the overflow check as done here.

Reply:
We can consider adding a wrapper function in `rcutils/allocator.h` to preform this check if you like, I already created a `reallocf` function along these lines:

- https://github.com/ros2/rcutils/blob/244fa98ad6421ed3d6295bd157ec4b73400fab52/include/rcutils/allocator.h#L129-L137
- https://github.com/ros2/rcutils/blob/244fa98ad6421ed3d6295bd157ec4b73400fab52/src/allocator.c#L100-L117

Otherwise, this check here seems fine.